### ISSUE TITLE
Link submodule via https:// not git@

### ISSUE THREAD

**Issue by loeken** [kryptco/kr#192]
Hi

just tried installing your project from the sources on a new arch template and i noticed i couldnt update the submodules. a peak inside the .git/config showed me that its linked via git@ so my terminal tried ssh connection to github and didnt have permission for this. Fix was to change it to https:// ( sed below )

so maybe change https://github.com/kryptco/kr/blob/master/.gitmodules to use https:// instead of git so its more convenient for non git users 

```
[user@archlinux ~]$ mkdir Test
[user@archlinux ~]$ cd Test/
[user@archlinux Test]$ git clone https://github.com/kryptco/kr
Cloning into 'kr'...
remote: Counting objects: 6537, done.
remote: Compressing objects: 100% (36/36), done.
remote: Total 6537 (delta 13), reused 26 (delta 7), pack-reused 6491
Receiving objects: 100% (6537/6537), 22.20 MiB | 4.94 MiB/s, done.
Resolving deltas: 100% (3491/3491), done.
[user@archlinux Test]$ cd kr/
[user@archlinux kr]$ git submodule update --init --recursive
Submodule 'sigchain' (git@github.com:kryptco/sigchain) registered for path 'sigchain'
Cloning into '/home/user/Test/kr/sigchain'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:kryptco/sigchain' into submodule path '/home/user/Test/kr/sigchain' failed
Failed to clone 'sigchain'. Retry scheduled
Cloning into '/home/user/Test/kr/sigchain'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:kryptco/sigchain' into submodule path '/home/user/Test/kr/sigchain' failed
Failed to clone 'sigchain' a second time, aborting

[user@archlinux kr]$ sed -i 's/git@github.com:kryptco\/sigchain/https:\/\/github.com\/kryptco\/sigchain\//' .git/config 
[user@archlinux kr]$ git submodule update --init --recursive
Cloning into '/home/user/Test/kr/sigchain'...
Submodule path 'sigchain': checked out 'dcb9a43b8d5e6b45d62bfc7a8ac9365d6ac78927'
Submodule 'libsodium' (https://github.com/jedisct1/libsodium) registered for path 'sigchain/libsodium'
Submodule 'openssl' (https://github.com/openssl/openssl) registered for path 'sigchain/openssl'
Cloning into '/home/user/Test/kr/sigchain/libsodium'...
Cloning into '/home/user/Test/kr/sigchain/openssl'...
Submodule path 'sigchain/libsodium': checked out '5ebc0d40295ac6c509aaf1d3ed5ded6bc2a77074'
Submodule path 'sigchain/openssl': checked out 'd4e4bd2a8163f355fa8a3884077eaec7adc75ff7'```

**Comment by OGKevin** [comment]
Kinda funny :D. Makes sense that they used ssh as thats what their product is all about. 

But AFAIK, if the repo is public then cloning via SSH should also work 🤔

**Comment by OGKevin** [comment]
Aha and im right

```
~ ➜ cd /tmp
/tmp ➜ git clone git@github.com:kryptco/kr.git
Cloning into 'kr'...
Kryptonite ▶ Requesting SSH authentication from phone
Kryptonite ▶ Phone approval required. Respond using the Kryptonite app
Kryptonite ▶ Success. Request Allowed ✔
remote: Counting objects: 6537, done.
remote: Compressing objects: 100% (36/36), done.
remote: Total 6537 (delta 13), reused 26 (delta 7), pack-reused 6491
Receiving objects: 100% (6537/6537), 22.20 MiB | 2.54 MiB/s, done.
Resolving deltas: 100% (3491/3491), done.
/tmp ➜ git clone git@github.com:kryptco/sigchain
Cloning into 'sigchain'...
Kryptonite ▶ Requesting SSH authentication from phone
Kryptonite ▶ Success. Request Allowed ✔
remote: Counting objects: 255, done.
remote: Compressing objects: 100% (11/11), done.
remote: Total 255 (delta 7), reused 10 (delta 3), pack-reused 241
Receiving objects: 100% (255/255), 561.92 KiB | 1.18 MiB/s, done.
Resolving deltas: 100% (30/30), done.
/tmp ➜
```

**Comment by OGKevin** [comment]
Via submodule also works. I suggest you check if your pub key is indeed correctly added to your GitHub account.

**Comment by loeken** [comment]
No problemo :). I also noticed that I seem to have missed a part on my initial post - updated the initial post

**Comment by kcking** [comment]
Thank you for reporting @loeken, we'll switch it to https

**Comment by OGKevin** [comment]
@kcking This issue can be closed in favour of 1ab262406b2c01265ab20ba9b8268ae90bb35f94 👍
